### ISSUE TITLE
Change AbstractTrainer constructor to retrieve ranker, error function and loss calculator

### ISSUE THREAD

**Issue by Kamulau** [LTR4L/ltr4l#256]
Right now, Abstract Trainer's constructor calls abstract methods constructRanker(), makeErrorFunc(), and makeLossCalc(), which are implemented in child classes.

I would like to eliminate these methods, and add in Abstract Trainer constructor:

[R extends Ranker]
AbstractTrainer(oldParams, R ranker, Error errorFunc, LossCalc lossCalc)
